src/mscorlib/src/Internal/Runtime/Augments/RuntimeThread.cs

```diff
@@ -186,6 +188,32 @@ public void DisableComObjectEagerCleanup()
         private extern bool JoinInternal(int millisecondsTimeout);
 
         public static void Sleep(int millisecondsTimeout) => Thread.Sleep(millisecondsTimeout);
+
+        [DllImport(JitHelpers.QCall)]
+        [SuppressUnmanagedCodeSecurity]
+        private static extern int GetOptimalMaxSpinWaitsPerSpinIterationInternal();
+
+        /// <summary>
+        /// Max value to be passed into <see cref="SpinWait(int)"/> for optimal delaying. This value is normalized to be
+        /// appropriate for the processor.
+        /// </summary>
+        internal static int OptimalMaxSpinWaitsPerSpinIteration
+        {
+            get
+            {
+                if (s_optimalMaxSpinWaitsPerSpinIteration != 0)
```

Comment:
>s_optimalMaxSpinWaitsPerSpinIteration [](start = 20, length = 37)

Looks this one can be converted to readonly field initialized with  GetOptimalMaxSpinWaitsPerSpinIterationInternal() so we can avoid checking 0 value.

Reply:
I didn't want do that since the first call would trigger the measurement that takes about 10 ms. Static construction of RuntimeThread probably happens during startup for most apps.